weatherTask/weatherman.py

```diff
@@ -1,151 +1,146 @@
-                    'temp_max': int(temp[1]),
-                    'temp_min': int(temp[3]),
-                    'humidity': int(temp[7])
-                })
-    return monthly_data
-
-def highest_lowest_temperature(d):
-    """Function to find highest and lowest temperature data"""
-    if not d:
-        return None
-
-    highest_temp = d[0]
-    lowest_temp = d[0]
-
-    for i in d:
-        if i['temp_max'] > highest_temp['temp_max']:
-            highest_temp = i
-        if i['temp_min'] < lowest_temp['temp_min']:
-            lowest_temp = i
-
-    return highest_temp, lowest_temp
-
-def highest_humidity(d):
-    """Function to get highest humidity data"""
-    if not d:
-        return None
-
-    high_humidity = d[0]
-
-    for i in d:
-        if i['humidity'] > high_humidity['humidity']:
-            high_humidity = i
-
-    return high_humidity
-
-def average_data(d):
-    """Function to get average temperature and humidity data"""
-    if not d:
-        return None
-
-    temp_max = 0
-    temp_min = 0
-    humidity = 0
-
-    for i in d:
-        temp_max += i['temp_max']
-        temp_min += i['temp_min']
-        humidity += i['humidity']
-
-    data_points = len(d)
-    avg_temp_max = temp_max / data_points
-    avg_temp_min = temp_min / data_points
-    avg_humidity = humidity / data_points
-
-    return avg_temp_max, avg_temp_min, avg_humidity
-
-def draw_bar_chart(chart_data):
-    """Function to get chart data"""
-    for d in chart_data:
-        print(f"{d['date']} {colored('+' * d['temp_max'], 'red')} {d['temp_max']}C")
-        print(f"{d['date']} {colored('+' * d['temp_min'], 'blue')} {d['temp_min']}C")
-
-def draw_singleline_bar_chart(chart_data):
-    """Function to get singleline chart data"""
-    for d in chart_data:
-        print(f"{d['date']} {colored('+' * d['temp_min'], 'blue')}"
-        f"{colored('+' * d['temp_max'], 'red')} {d['temp_min']}C - {d['temp_max']}C")
+            lines = file.readlines()[1:-1]
+            report_file = csv.DictReader(lines)
+            for row in report_file:
+                fields = ["Max TemperatureC", "Min TemperatureC", "Max Humidity"]
+                if any(row[field] == '' for field in fields):
+                    continue
+                row['Max TemperatureC'] = int(row['Max TemperatureC'])
+                row['Min TemperatureC'] = int(row['Min TemperatureC'])
+                row['Max Humidity'] = int(row['Max Humidity'])
+                row['PKT'] = datetime.strptime(row['PKT'], '%Y-%m-%d')
+                data.append(row)
+        return data
+
+    def data_filepath(self, file_type):
+        """method to get data"""
+        data = []
+        if file_type == 'yearly':
+            for index in self.month:
+                file_path = os.path.join(self.folderpath, f"lahore_weather_{self.year}_{index}.txt")
+                weather_data = self.read_weather_data(file_path)
+                if weather_data:
```

Comment:
why is this if condition required?

Reply:
sir some of the files are not available, so this condition checks if the year data is available for every month which in this case is false because year 1996 has only one month(Dec) so when it checks for the 1st month file like Jan it throws an error "NoneType object is not iterable"